### ISSUE TITLE
Document the best way to use Emulsify with client work

### ISSUE THREAD

**Issue by a-fro** [fourkitchens/emulsify#38]
To start, I _love_ what you guys are doing here! I presented Emulsify and PatternLab at our recent Ithaca Drupal meetup. Great work!

I'm trying to plan for the best way to build from the work here when using it on a project. I'm guessing/hoping that you're thinking about this at FourKitchens as well. We could basically fork it from a given moment in time as a custom theme, and then do the work of pulling  in relevant upstream changes over time. I've also been considering creating a custom theme that becomes the base theme of Emulsify, but that would only pull the relevant styles into Drupal, and not PatternLab. Finally, I've been wondering about the possibility of creating some scripts and or tasks (using bash or gulp) that could pull upstream Emulsify changes and additions to both Drupal Twig templates and the pattern templates. 

What are you all thinking about the best way to use Emulsify/PatternLab for client work?

**Comment by evanmwillhite** [comment]
Thanks @a-fro! We did consider this when building and it still comes up from time-to-time internally. In fact, thanks for starting this conversation publicly, as I'd be interested to get feedback from anyone using it outside of Four Kitchens. 

So far, we have treated it as a starterkit - something used as-is and that should be changed as needed. The reasoning behind this is that the only functionality provided (pre-built components) should be removed/tweaked based on the specific project's need. Pattern Lab itself can be updated separately at any time. Unlike a base theme, there's no functionality that needs to be tucked away for separation of concerns. As far as upstream changes are concerned, these would likely impact any existing components you have built, so it would be a manual process to update them anyway. 

In terms of our current workflow (I think you will know all this but in case it helps anyone reading this), we use Emulsify as a standalone prototyping tool to create all our markup, Sass/CSS, and Javascript. That is then dropped right into the Drupal project and used as the theme. From there, it's only a matter of using/adding the correct Drupal templates in the `/templates` directory and using Twig's include, extends or embed functions to pull in the markup from the `/components` Twig files. The CSS file that all the Sass is compiled into is already referenced as a theme library and the component Javascript can be added into the Drupal templates as needed (see docs [here](https://github.com/fourkitchens/emulsify/blob/master/docs/drupal-components.md#javascript-in-drupal)). So, all markup, CSS and Javascript are kept in the `/components` directory and are merely referenced from Drupal (or whatever system is being used). There is a minimal amount of project-specific CSS that is only used for Pattern Lab, but it's so minimal (like a few lines of CSS), that we tend to keep that alongside the other CSS (see an [example](https://github.com/fourkitchens/emulsify/blob/master/components/_patterns/00-base/layouts/grid/_01-grid-item.scss#L39-L45) here).

Hopefully this helps, and let me know if you still have questions. Thanks again for your excitement and support!
